File: bkmonitor/webpack/src/monitor-pc/router/scenes/rum.ts
/*
 * Tencent is pleased to support the open source community by making
 * 蓝鲸智云PaaS平台 (BlueKing PaaS) available.
 *
 * Copyright (C) 2017-2025 Tencent.  All rights reserved.
 *
 * 蓝鲸智云PaaS平台 (BlueKing PaaS) is licensed under the MIT License.
 *
 * License for 蓝鲸智云PaaS平台 (BlueKing PaaS):
 *
 * ---------------------------------------------------
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
 * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
 * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
 * the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
 * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
 * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
import * as rumAuth from '../../pages/rum/authority-map';

import type { RouteConfig } from 'vue-router';

const Rum = () => import(/* webpackChunkName: 'Rum' */ '../../pages/rum/rum');
export default [
  {
    path: '/trace/rum',
    name: 'rum',
    components: {
      noCache: Rum,
    },
    meta: {
      title: 'RUM',
      navId: 'rum',
      navClass: 'rum-nav',
      noChangeLoading: true,
      noNavBar: true,
      needClearQuery: true,
      route: {
        parent: 'scenes',
      },
      authority: {
        map: rumAuth,
        page: rumAuth.VIEW_AUTH,
      },
    },
  },
  {
    path: '/trace/rum/app/:appName/config',
    name: 'rumAppConfig',
    components: {
      noCache: Rum,
    },
    meta: {
      title: 'RUM应用配置',
      needBack: true,
      navId: 'rum',
      authority: {
        map: rumAuth,
        page: rumAuth.VIEW_AUTH,
      },
      route: {
        parent: 'rum',
      },
      noNavBar: true,
    },
  },
] as RouteConfig[];

File: bkmonitor/bkmonitor/data_source/utils/rum.py
"""
Tencent is pleased to support the open source community by making 蓝鲸智云 - 监控平台 (BlueKing - Monitor) available.
Copyright (C) 2017-2025 Tencent. All rights reserved.
Licensed under the MIT License (the "License"); you may not use this file except in compliance with the License.
You may obtain a copy of the License at http://opensource.org/licenses/MIT
Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
"""

from dataclasses import dataclass


@dataclass(frozen=True)
class RUMAppTarget:
    """RUM 应用维度目标，承载业务 ID 与应用名称"""

    bk_biz_id: int
    app_name: str


@dataclass(frozen=True)
class RUMDatasourceTarget:
    """Trace 数据源查询目标，表示一条`table_id -> APM 应用`的绑定关系"""

    table_id: str
    app: RUMAppTarget

    @classmethod
    def build(cls, bk_biz_id: int, app_name: str, table_id: str) -> "RUMDatasourceTarget":
        return cls(table_id=table_id, app=RUMAppTarget(bk_biz_id=bk_biz_id, app_name=app_name))
